tools/monorepo-utils/src/core/git.ts

```diff
@@ -83,7 +83,7 @@ export const cloneRepo = async ( repoPath: string ) => {
 	mkdirSync( folderPath, { recursive: true } );
 
 	const git = simpleGit( { baseDir: folderPath } );
-	await git.clone( repoPath, folderPath );
+	await git.clone( repoPath, folderPath, { '--depth': 1 } );
```

Comment:
@samueljseay I added this option to significantly reduce the amount of time downloading the repo. It only gathers the latest commit. I'm pretty sure none of our tools should require more than one commit. What do you think? Is it a safe assumption? I also added the same change below to `sparseCheckoutRepo`.

Reply:
@psealock its a great idea. My only concern is that it won't work for situations where we need to check out specific commits to do diffs (without having to reclone), such as code analyzer. I think it'd be low effort to do either of:

1. write a cloneShallow function **or**
2. add a cloneOptions argument to the function (perhaps it could also default to `{ depth: 1 }`, make sure the function docs are explicit about this.)

I found when doing git operations a lot in tools, not having to dig into functions to make assumptions about what was happening was important as it often lead to "wtf" moments. To minimize this kind of pain I think we should be explicit in function name or documentation *or* we should make it configurable.

What do you think though?

Reply:
Thanks for the suggestion @samueljseay. I have introduced `sparseCheckoutRepoShallow` and `cloneRepoShallow` while updating their underlying functions to accept options.